temba/orgs/views.py

```diff
@@ -1260,6 +1276,22 @@ def pre_save(self, obj):
             analytics.track(self.request.user.username, 'temba.org_configured_webhook')
 
             obj.webhook_events = webhook_events
+
+            webhook_data = {}
+            if data['webhook']:
+                webhook_data.update({'url': data['webhook']})
+                webhook_data.update({'method': 'POST'})
+
+            if data['headers']:
+                pairs = []
+                for n in xrange(len({k for k in data['headers'] if 'key' in k})):
+                    pair = {k: v for k, v in data['headers'].items() if k.startswith('header_' + str(n + 1))}
+                    pairs.append(pair.items())
+                as_dict = dict([(p[0][1], p[1][1]) for p in pairs])
+                webhook_data.update({'headers': as_dict})
```

Comment:
Man this just looks super awkward in the few places I've seen it..  

If I understand things correctly we are just trying to go from header_n_key and header_n_value to a dictionary?  What about something like the below in the form to turn this into the dict.. 

Also any reason the Form isn't just taking care of the dict() transformation so you don't need to do this mapping twice?

```python
idx = 0
headers = dict()
key = 'header_%d_key' % idx
value = 'header_%d_value' % idx

while key in data:
  if data.get(value, ''):
    headers[data[key]] = data[value]

  idx += 1
  key = 'header_%d_key' % idx
  value = 'header_%d_value' % idx
```